pkg/pipeline/pipeline.go

```diff
@@ -281,9 +281,19 @@ func (p *Pipeline) messageWatch(msg *gst.Message) bool {
 		}
 
 		p.Logger.Debugw("EOS received, stopping pipeline")
-		if p.Info.Status == livekit.EgressStatus_EGRESS_ACTIVE {
-			p.Info.Status = livekit.EgressStatus_EGRESS_ENDING
-		}
+		p.closeOnce.Do(func() {
```

Comment:
Can this block be refactored in a subroutine since we have basically the same code in SendEOS?